kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs

```diff
@@ -18,23 +18,54 @@ impl<'tcx> GotocCtx<'tcx> {
         let label: String = self.current_fn().find_label(&bb);
         // the first statement should be labelled. if there is no statements, then the
         // terminator should be labelled.
+        let check_coverage = self.queries.check_coverage;
         match bbd.statements.len() {
             0 => {
                 let term = bbd.terminator();
-                let tcode = self.codegen_terminator(term);
-                self.current_fn_mut().push_onto_block(tcode.with_label(label));
+                if check_coverage {
```

Comment:
I won't lie... not a big fan of all these `if check_coverage` in this function. It can be in a follow up PR, but I think wrapping the coverage handling logic will help a lot with maintenance and readability here.

Reply:
I'd like to know how you'd do this and take care of it now. I've [refactored this code a little bit](https://github.com/model-checking/kani/pull/2609/commits/823d72a42896a6d34e909710982448384d598e4f), but I'm not a fan of the code neither.

Reply:
Sure. Let's take this offline since I don't think it's a blocker for this PR.